pandas/indexes/base.py

```diff
@@ -629,7 +629,7 @@ def copy(self, name=None, deep=False, dtype=None, **kwargs):
             name = name or deepcopy(self.name)
         else:
             new_index = self._shallow_copy()
-            name = self.name
```

Comment:
Although admittedly nicer to use `or`, this will stop people using `0` or an empty string as a name.